docs/ads.md

```diff
@@ -0,0 +1,458 @@
+# Ads
+
+The Kaltura TV Platform Player supports the addition of 3rd party monetization solutions via its plugin system. To allow easy integration and to support playback and analytics using any 3rd party solution, the player defines an advertizing framework using APIs and a standard event scheme.
+
+Any plugin that adds support for ads is expected to follow the controller, timeline, and events scheme.
+This document details the ads controller, which allows controlling ad-related operations, the ad timeline (which explains how a typical order of events looks like), and the ad events and their associated payload.
+
+## Ads Controller
+
+To query the player regarding the ad's state in the current playback, you can use the ads controller by referencing `player.ads`.
+The ads controller is responsible for tracking and communicating with the active ads plugin, as well as for collecting and saving the current ads state and data.
+![ad-controller](images/ads-controller.jpg)
+
+> A complete reference to the AdsController can be found [here](#AdsController).
+
+#### Examples:
+
+##### Know when a scheduled ad break will play
+
+```js
+const layout = player.ads.getAdBreaksLayout();
+console.log(layout); // [0, 15, -1]
+```
+
+##### Check if all ads are completed
+
+```js
+if (player.ads.allAdsCompleted) {
+  // do something
+}
+```
+
+##### Check if this is the middle of an ad break
+
+```js
+if (player.ads.isAdBreak()) {
+  // do something
+}
+```
+
+##### Get the current ad break data
+
+```js
+if (player.ads.isAdBreak()) {
+  const adBreak = player.ads.getAdBreak();
+  // do something
+}
+```
+
+##### Get the current ad data
+
+```js
+if (player.ads.isAdBreak()) {
+  const ad = player.ads.getAd();
+  // do something
+}
+```
+
+## Ad Events Timeline
+
+During ad playback, there are events that get triggered to indicate the current ad state or state changes.
+Some events are life cycle indicators and will always get triggered, while others are reactive to specific actions -- such as the user clicking the ad -- and may or may not be triggered.  
+
+In the following diagram you can observe the events timeline from the point that an ad break starts to the point that the ad break ends.
+
+![ad-events-timeline](images/ad-events-timeline.jpg)
+
+If you want to know when playback has started/ended -- whether the playback includes ads or not -- use the following events:
+
+- `PLAYBACK_START` - Triggered on the first play request for playback. If a pre-roll ad is about to play, this event will be triggered before the pre-roll. If no pre-roll ad is about to play, this event will be triggered before any other content event including `FIRST_PLAY`.
+
+- `PLAYBACK_ENDED` - Triggered when playback is ended. If a post-roll ad is about to play, this event will be triggered after post-roll. If no post-roll ad is about to play, this event will be triggered after any other content event including `ENDED`.
```

Comment:
I change the first part of both pre-roll and post-roll to "is about to play" - verify that you are ok with this change.